### ISSUE TITLE
Declaration of 'RDFterm-equal' in Sec.17.4.1.7 inconsistent with the way it is used in Sec.17.3

### ISSUE THREAD

**Issue by hartig** [w3c/sparql-query#18]
The section that defines `RDFterm-equal` (Section 17.4.1.7) begins with the following declaration.
```
xsd:boolean  RDF term term1 = RDF term term2
```
So, it doesn't actually look like a function declaration, and it is inconsistent with how `RDFterm-equal` is used in the operator table in Section 17.3. To fix this inconsistency, I propose to change the aforementioned declaration to look as follow.
```
xsd:boolean RDFterm-equal (RDF term term1, RDF term term2)
```

**Comment by afs** [comment]
`RDFterm-equal` is only used for `=` and `!=`

See also `||` and `&&` -- also in the operator table. (They are not strict functions.)

https://www.w3.org/TR/sparql11-query/#func-logical-or
https://www.w3.org/TR/sparql11-query/#func-logical-and

The declaration of `RDFterm-equal` is written that way so that there isn't the possibility of writing `FILTER ( RDF-equal(?x, ?y) )`.

Unlike `||` and `&&`, this coudl be a function (the other two aren't pure functions).

So we could add `RDFterm-equal` as proposed - it is a function - though it does imply a grammar addition.
Or we could add text to say explain the situation.
Or change the declaration and also say "you can't write this in functional form".

**Comment by hartig** [comment]
> So we could add RDFterm-equal as proposed - it is a function - though it does imply a grammar addition.
> Or we could add text to say explain the situation.
> Or change the declaration and also say "you can't write this in functional form".

I would prefer the last of these three options.

**Comment by afs** [comment]
In XPath/XQuery, the same situation occurs with operators: the `op:` namespace does not provide callable functions.
https://www.w3.org/TR/xpath-functions-31/#func-numeric-add

"Defines the semantics of the "+" operator when applied to two numeric values"

**Comment by hartig** [comment]
I will create a PR that implements the third option using a similar statement as the one that you quoted from XPath. However, I will wait until PR #17 is merged to not interfere with the way how that PR changes the section about `RDFterm-equal`.

In the PR that I will create, I will also implement the same type of fix for Sections 17.4.1.5 and 17.4.1.6, which are about the functions `logical-or` and `logical-and`, respectively, and which have the same issue (the signature of these functions in these sections is inconsistent with how these functions are used in the operator mappings table in Sec.13.7).
